### DISCUSSION TITLE
Could Testbench load my service provider from the composer.json file?

### DISCUSSION THREAD

**Discussion by christophrumpel** [orchestral/testbench#421] [Ideas]
**The "Issue"**

I was working on my package and writing tests. I added my package's service provider to my `TestCase` file using `getPackageProviders` from Testbench. All tests were passing; all seemed good. But I hadn't added my service provider to my `composer.json` file under `extras`. So in real, the provider was not loaded in a Laravel application.

**Idea**

Could Testbench check the composer.json file too?

**Comment by christophrumpel** [comment]
I only found this, but if I'm correct, this only applies to other packages I use?

https://packages.tools/testbench/the-basic#package-auto-discovery

**Reply by crynobone** [reply]
Once you run `php vendor/bin/testbench package:discover` it will generate `boottrap/cache/services.php` to the stub skeleton and you can use the above helper to autoload the packages on your tests.

**Reply by christophrumpel** [reply]
🤔 thanks need to try this
